doc/api/http.md

```diff
@@ -3206,6 +3206,41 @@ server.on('request', (request, res) => {
 server.listen(8000);
 ```
 
+## `http.createSimpleServer([directory[, port[, host]]])`
+
+<!--YAML
+added: REPLACEME
+-->
+
+> Stability: 0 - Experimental
+
+* `directory` {string|URL} Root directory from which files would be serve.
+  **Default:** `process.cwd()`.
+* `port` {number}
+* `host` {string} **Default:** `'localhost'`
+
+* Returns: {http.Server}
+
+Start a TCP server listening for connections on the given `port` and `host`, and
+serve statically local files, using `directory` as the root. Please note that
+when specifying a `host` other than `localhost`, you are exposing you local file
+system to all the machines that can connect to your computer.
+
+If `port` is omitted or is 0, the operating system will assign an arbitrary
+unused port, which will be output the standard output.
+
+Also accessible via `require('node:http/server')`.
+
+```bash
+node -r node:http/server # starts serving the cwd on a random port
+
+# To start serving on the port 8080 using /path/to/dir as the root:
+node -r node:http/server /path/to/dir --port 8080
```

Comment:
Not really a fan of building in a preload like this. Someone can easily use the `-pe` syntax to achieve this if they wanted, e.g. `node -pe "require('http').createStaticServer(...)`

Reply:
One reason to use `-r` is you can get autocomplete when selecting the root directory, and it's easier to offer shorter versions of flags (i.e. I'd rather type `node -r http/server . -p 8080 -h 0.0.0.0` than `node -e 'http.createStaticServer({port: 8080, host: '0.0.0.0'})'`). Currently this PR offers both syntax, so I guess I should also document that, and folks can chose the syntax they prefer, wdyt?